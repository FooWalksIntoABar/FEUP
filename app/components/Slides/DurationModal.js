import React from 'react'
import { StyleSheet, View, Text, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'

export class DurationModal extends React.Component {
  constructor () {
    super()
    this.state = {
      selectedItem: 1,
      itemsList: ['1', '2', '3']
    }
  }

  componentDidMount () {
  }

  initItemsList = () => {
    let filledListValues = []
    for (let i = 0; i < 60; i++) {
      filledListValues.push(i)
    }
    let newItems = this.state.itemsList.concat(filledListValues)
    this.setState({
      itemsList: newItems
    })
    console.log('ahahah' + this.state.itemsList)
  }

  onPickerSelect = (index) => {

  }
  render () {
    return (
      <View style={{ padding: 10 }}>
        <Text style={styles.centeredElement}>Escolha a duração da atividade</Text>
        <View style={[styles.centeredElement, { justifyContent: 'flex-end' }]}>

          <TouchableOpacity onPress={this.props.closeModalCallback}>
            <Text>EXIT</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  centeredElement: {
    alignSelf: 'center'
  }
})

DurationModal.propTypes = {
  closeModalCallback: PropTypes.func.isRequired
}
