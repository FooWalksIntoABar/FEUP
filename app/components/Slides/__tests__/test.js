import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { DurationModal } from '../DurationModal'

configure({ adapter: new Adapter() })

describe('Slide components', () => {
  it('renders DurationModal correctly', () => {
    const wrapper = shallow(<DurationModal
      closeModalCallback={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.instance().onPickerSelect(0)
    wrapper.instance().initItemsList()
    wrapper.unmount()
  })
})
