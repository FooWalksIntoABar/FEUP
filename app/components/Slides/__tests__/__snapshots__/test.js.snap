// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide components renders DurationModal correctly 1`] = `
<Component
  style={
    Object {
      "padding": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignSelf": "center",
      }
    }
  >
    Escolha a duração da atividade
  </Text>
  <Component
    style={
      Array [
        Object {
          "alignSelf": "center",
        },
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[MockFunction]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        EXIT
      </Text>
    </TouchableOpacity>
  </Component>
</Component>
`;
